### PR TITLE
Create Stratum-tools package and Docker image

### LINKED ISSUE #603: Publish stratum-tools debian package and Docker container
* Stratum replay
* Stratum pipeline builder
* gNMI CLI
* Chassis Config migrator
* ONLP CLIs
* ...

### PR DESCRIPTION
**Pull request by pudelkoM**
This PR creates the Stratum-tools Debian package and Docker image. It contains various tools and binaries related to Stratum, but not the switch runtime itself.

The new Docker container is published as `stratumproject/stratum-tools`.

- bf_pipeline_builder
- stratum_dummy
- chassis_config_migrator
- phal_cli
- ~onlp_cli~
- stratum_stub
- procmon_main
- gnmi_cli
- stratum_replay

This also deprecates the following docker images:
 - `stratumproject/stratum-replay`
 - `stratumproject/stratum-bf-pipeline-builder`
 - `stratumproject/chassis-config-migrator`
 - `stratumproject/gnmi-cli`

They will remain available for some time, but are not updated anymore.

Closes #603

### CODE REVIEW

**Comment by pudelkoM** [comment]
@ccascone we talked about a potential stratum-tools package recently, but I forgot what tool we had in mind. If you remember, please add here.

File: stratum/tools/BUILD [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        "//stratum/hal/bin/dummy:stratum_dummy",
+        "//stratum/hal/config:chassis_config_migrator",
+        "//stratum/hal/lib/phal:phal_cli",
+        "//stratum/hal/lib/phal/onlp:onlp_cli",
+        "//stratum/hal/stub/embedded:stratum_stub",
+        "//stratum/procmon:procmon_main",
+        "//stratum/tools/gnmi:gnmi_cli",
+        "//stratum/tools/stratum_replay",
+    ],
+    extension = "tar.xz",
```
Comment (pudelkoM) [inline stratum/tools/BUILD:19]: This gives us the best compression ratio and makes the whole package just 8.3M small.

File: stratum/tools/Dockerfile.stratum_tools [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ADD ./gnmi_cli /usr/local/bin
+ADD ./onlp_cli /usr/local/bin
+ADD ./phal_cli /usr/local/bin
+ADD ./procmon_main /usr/local/bin
+ADD ./stratum_dummy /usr/local/bin
+ADD ./stratum_replay /usr/local/bin
+ADD ./stratum_stub /usr/local/bin
+
+ENTRYPOINT ["bash"]
+CMD ["ls -l /usr/local/bin"]
```
Comment (pudelkoM) [inline stratum/tools/Dockerfile.stratum_tools:20]: Since this is a collection, I figured we just print a list of the available tools.

**Review by ccascone — APPROVED** [review]
LGTM. I cannot think of anything else that is not already mentioned in your list.

**Pull request merged**